### DISCUSSION TITLE
CSS/JS auto versioning based on last git commit hash

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#3943] [Extensions & Modules]
Hi everybody,
if you use long browser cache lifetime (to make google happy) and you have to flush your CDN every time you modify one of your CSS/JS then this module could be for you.

https://github.com/fballiano/openmage-cssjs-versioning

I've used this solution for many years for all my customers but I never packaged it in a "standardized" way until now.

Hope this may help somebody out there.

**Quick description:**
Adds ?v= to all your CSS/JS and the value of v is the last git commit hash (or a timestamp if the git approach fails).

This way CDN updates and also browsers will download the new version of the files.

**Comment by ma4nn** [comment]
Nice👍🏻 I use a very similar solution, but based on a simple time-based version string. This eliminates the need for .git repository in production and the result is nearly the same if cache is not flushed too often.

**Reply by fballiano** [reply]
you're checkin modification time of some file?

**Reply by ma4nn** [reply]
no I simply save the current time as version string in cache if it is not existent and append that cache entry to assets

**Reply by fballiano** [reply]
ah okok, I prefer to have the "version" not change if not necessary and not being tied to the flushing of the cache.

but still a totally valid approach

**Reply by fballiano** [reply]
I've added a fallback to timestamp if the git thing fails ;-)

**Comment by kiatng** [comment]
I use this: https://github.com/jreinke/magento-suffix-static-files

**Comment by pquerner** [comment]
Having .git folders on prod seems scary. Why not `mtime` ?

**Reply by fballiano** [reply]
seems scary but it's not, it also allows for a "git status" and keep an eye to make sure nobody changed any file anywhere. it also allow not to change the calculated version number if there's no change in the software

anyway the module now supports time too if there's no git folder.

**Reply by kiatng** [reply]
Just use timestamp for universality and code reduction.

**Reply by fballiano** [reply]
noup, having the version number constantly changing (without any change in the code) is simply wrong, it stresses the CDN for no reason and it renders the browsers incapable of using their cache effectively.

**Reply by luigifab** [reply]
No use git folder in your production env and commit the vendor directory and use the public dir!

There were some discussions about the public dir, I don't retrieve them.

**Reply by fballiano** [reply]
the module supports both
